src/python/pants/backend/python/macros/python_requirements.py

```diff
@@ -110,9 +152,16 @@ async def generate_from_python_requirement(
             description_of_origin=f"{generator}'s field `{PythonRequirementsSourceField.alias}`",
         ),
     )
-    requirements = parse_requirements_file(
-        digest_contents[0].content.decode(), rel_path=requirements_full_path
-    )
+    if requirements_rel_path.endswith("toml"):
```

Comment:
Its a bit contrived, but also easy to avoid: "requirements.transfertoml" where toml is read as "to ML" and these requirements are for some sort of data pipeline tool.
How about using `x.endswith(".toml")` or `os.path.splitext(x)[1] == ".toml"`?